src/scripts/app.ts

```diff
@@ -48,51 +41,56 @@ const setDifficulty = () => {
 	if (selectedInput) {
 		const selectedValue = selectedInput.value;
 		localStorage.setItem('selectedInput', selectedValue);
+		checkChoseValue();
 	} else {
-		showToast('warning', 'Please select your difficulty level', {
-			position: 'left',
-		});
+		showToast('warning', 'Please select your difficulty level', {});
 	}
 };
 
-const setGrid = () => {
+async function checkChoseValue() {
 	const chosenValue = localStorage.getItem('selectedInput') as keyof GameLevels;
 
 	const chosedValue = gameLevels[chosenValue] || gameLevels.easy;
 	console.log(chosedValue);
 
-	addItems();
-};
+	await fetchData();
+	await addItems(chosedValue, imageArray);
+}
 
 async function fetchData() {
-	// await axios.get(URL).then((res) => {
-	// 	console.log(res.data);
-	// });
 	try {
-		const result = await axios.get(URL);
-		showToast('success', 'Data fetched successfully');
-		console.log(result);
-	} catch (e) {
-		showToast('error', 'Something went wrong, try again later');
+		unsplash.search
+			.getPhotos({ query: 'star-wars', orientation: 'portrait', perPage: 30 })
+			.then((result) => {
+				console.log(result);
+
+				if (!result.response?.results) {
+					showToast('error', 'Cannot get API data');
+				} else {
+					result.response.results.forEach((item) => imageArray.push(item));
```

Comment:
Dokladnie:)
Czyli rozwiązania były dwa;
- Zrobienie `const imagesArray: Basic[] = []`;
- A pozniej: `result.response.results.forEach((item) => imageArray.push(item));`
Bo wtedy nie przypisujesz nowej wartosci imagesArray, tylko po prostu pushujesz do tablicy, wiec const jest okej

A jezeli chcesz inaczej to mozesz zrobic:
- `let imagesArray: Basic[] = []`
- imagesArray = result.response.results;